app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt

```diff
@@ -298,6 +298,16 @@ fun ConversationScreen(
             messageComposerStateHolder.messageCompositionHolder.value.updateQuote(compositionState.quotedMessage)
         }
     }
+    // add mentions when selectedMentions is not empty in MessageDraft
+    LaunchedEffect(messageDraftViewModel.state.value.selectedMentions) {
+        val compositionState = messageDraftViewModel.state.value
+        if (compositionState.selectedMentions.isNotEmpty()) {
+            messageComposerStateHolder.messageCompositionHolder.value.setMentions(
+                compositionState.draftText,
+                compositionState.selectedMentions.map { it.intoMessageMention() }
+            )
+        }
+    }
```

Comment:
mentions should be set with with the whole draft and not independent from the text it self, otherwise we can risk having the text and the mention out of sync causing out of bound exceptions